.github/workflows/build.yaml

```diff
@@ -59,7 +60,21 @@ jobs:
       - name: Setup Maven repository
         id: configure-maven-settings
         run: cp $GITHUB_WORKSPACE/.maven.settings.xml $HOME/.m2/settings.xml
-      
+
+      - name: Determine Architecture
+        id: detect_architecture
+        run: echo "::set-output name=arch::$(uname -m)"
+
+      - name: Set Dockerfile.amd based on Architecture
+        run: |
+          if [ "${{ steps.detect_architecture.outputs.arch }}" == "aarch64" ]; then
+            echo "Using ARM architecture Dockerfile"
+            export DOCKERFILE="Dockerfile.arm"
+          else
+            echo "Using AMD architecture Dockerfile"
+            export DOCKERFILE="Dockerfile.amd"
```

Comment:
see if this is the standard way of doing this
check how mongo or any other docker file management for different environments

Reply:
had a chat with Ujjwal, he told me to stick to this